types/part_set.go

```diff
@@ -270,9 +270,12 @@ func (ps *PartSet) Total() uint32 {
 }
 
 func (ps *PartSet) AddPart(part *Part) (bool, error) {
+	// TODO: remove this? would be preferable if this only returned (false, nil)
+	// when its a duplicate block part
 	if ps == nil {
 		return false, nil
 	}
```

Comment:
Does leaving this check here mean that a metric can be incremented on the wrong condition, which also "should not occur" in correct code and so it's hard to debug when it does occur?

Reply:
I added a similar comment in the original PR. I think we should change line 276 to return a non-nil error. But it is orthogonal to the original PR (and to this one)

Reply:
Filed as #920.